WalletWasabi.Fluent/Helpers/StartupHelper.cs

```diff
@@ -48,5 +55,33 @@ private static void ModifyRegistry(bool runOnSystemStartup, string pathToExeFile
 				key.DeleteValue(nameof(WalletWasabi), false);
 			}
 		}
+
+		private static void StartOnMacStartUp(bool runOnSystemStartup)
+		{
+			if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+			{
+				throw new InvalidOperationException("Running osascript can only be done on macOS.");
+			}
+
+			ProcessStartInfo processInfo = new()
+			{
+				UseShellExecute = true,
+				WindowStyle = ProcessWindowStyle.Hidden,
+				FileName = "/bin/bash",
+				CreateNoWindow = false,
+				Arguments = runOnSystemStartup ? AddArguments : DeleteArguments
+			};
+
+			using Process? process = Process.Start(processInfo);
+
+			int timeout = 5000;
+
+			process?.WaitForExit(timeout);
+
+			if (process?.ExitCode != 0)
+			{
+				throw new InvalidOperationException($"Couldn't add Wasabi Wallet to the Login Items. Exit code: {process?.ExitCode}");
+			}
```

Comment:
Why not simply add `if (process is not null) {  /* contents here */ }`?

Reply:
I would be surprised though if `process` is ever null here though because it may be null in very specific situations I think.

Reply:
I would rather not nest everything inside an `if (process is not null) ` because as you said and as i read it returns `null` in very specific situations. Maybe I should add a `null` check and throw exception if that happens.

Reply:
What bothers me more is that on CI the process exits with `Exit code 1`... I need to grab a Mac and test this PR further more.

Reply:
It would be helpful to read the stderr and stdout of that process because it most likely reports an error and it's only hidden.

Reply:
Yes, I will add it to the `ShellExecAsync` when we log out the `command` and the `exit code` in this PR or in another one.